editor/src/clj/editor/workspace.clj

```diff
@@ -467,14 +467,15 @@ ordinary paths."
 (def ^:private java-resource-path "templates/template.")
 
 (defn- get-template-resource [workspace resource-type]
-  (let [resource-path (:template resource-type)]
+  (let [resource-path (:template resource-type)
+        ext (:ext resource-type)]
     (or
       ;; default user resource
-      (find-resource workspace (some->> resource-type :ext (str default-user-resource-path)))
+      (when ext (find-resource workspace (str default-user-resource-path ext)))
       ;; editor resource provided from extensions
       (when resource-path (find-resource workspace resource-path))
       ;; java resource
-      (io/resource (some->> resource-type :ext (str java-resource-path))))))
+      (when ext (io/resource (str java-resource-path ext))))))
```

Comment:
`:ext` is definitely not nil. We should either remove nil checks here (i.e. replace `some->>` with `->>`) because they were added by mistake or figure out why there were nullability checks (I don't know if `resource-type` could be `nil` here).

Reply:
I think I understand now. Yes, the resource type may be `nil` if we create a file using `New File` and type an unregistered extension, e.g. `new file -> foo.bleh`. See `handler/defhandler :new-file :global`'s `run` function that invokes `workspace/template`. It would be nice for the editor to still use a template `/templates/default.bleh` in this case, but that would require more refactoring — we would need to pass an extension, and the function would look up resource type by this extension to see if there is an explicit template; and if there isn't, then it would look up fallbacks.

Reply:
It feels like a small fraction out of a % of use cases. Do we want to solve it now as part of this issue?
Or maybe it's enough to make sure it's not nil:
```clj
(defn- get-template-resource [workspace resource-type]
  (when resource-type
    (let [resource-path (:template resource-type)
          ext (:ext resource-type)]
      (or
        ;; default user resource
        (find-resource workspace (str default-user-resource-path ext))
        ;; editor resource provided from extensions
        (when resource-path (find-resource workspace resource-path))
        ;; java resource
        (io/resource (str java-resource-path ext))))))

Reply:
I think so, it's a fairly small change to accept an extension instead of a resource type.

Reply:
it can't be extension
it should be a full path
because native extensions provide full path
maybe I'll have time to take a look into it next week

Reply:
when you have a file extension and a workspace, you can look up the resource type using `get-resource-type`